### PR TITLE
test: refactor 'set' tests to AVA

### PR DESCRIPTION
**Pull request by nfischer**
This is an automated PR, so double check that I got all the test cases. I'll follow up once I've verified that this translates everything correctly.

### CODE REVIEW

File: ava-test/set.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  shell.config.silent = true;
+  shell.cp('-r', 'resources', t.context.tmp);
+});
+
+test.afterEach.always(t => {
+  shell.rm('-rf', t.context.tmp);
+});
+
+const nodeVersion = process.versions.node.split('.').map(str => parseInt(str, 10));
+const uncaughtErrorExitCode = (nodeVersion[0] === 0 && nodeVersion[1] < 11) ? 8 : 1;
```
Comment (freitagbr) [inline ava-test/set.js]: Move these up a few lines with the other `const` declaration.

Comment (nfischer) [inline ava-test/set.js]: This is some legacy stuff. More of our stuff breaks on v0.10, and even this code doesn't work on v0.8, so I'm just taking it out. The value for v0.11+ is `1`.

**Comment by nfischer** [comment]
@freitagbr PTAL

**Comment by freitagbr** [comment]
LGTM

**Comment by nfischer** [comment]
@freitagbr can you approve the changes?

**Pull request merged**